index.js

```diff
@@ -1,88 +1,239 @@
+            // Note: this freezes the @eslint/config-inspector page when expanding the `perfectionist` settings,
+            // as it generates too many characters.
+            alphabet: Alphabet.generateRecommendedAlphabet()
+              .sortByCharCodeAt()
+              // Nested imports come before the parent imports,
+              // i.e. `@/components/foo/nested.vue` before `@/components/foo.vue`
+              // TODO: do we really need this?
+              .placeCharacterBefore({ characterBefore: '/', characterAfter: '.' })
+              // `@/components/foo/nested.vue` comes before `@/components/foo-bar.vue`
+              .placeCharacterBefore({ characterBefore: '/', characterAfter: '-' })
+              .getCharacters()
+          }
+        },
+        rules: {
+          'perfectionist/sort-imports': [
+            'error',
+            {
+              type: 'custom',
+              ignoreCase: false,
+              newlinesBetween: 'ignore',
+              groups: [
+                'builtin',
+                'subpath',
+                'aliased-external',
+                'external',
+                'internal',
+                'aliased-internal',
+                'parent-composables',
+                'parent-components',
+                'parent',
+                'sibling',
+                'index'
+              ],
+              // The default is `['^~/.+', '^@/.+']`
+              // But we want to group `~*/** */` before other external imports,
+              // and `@/**/*` after normal internal imports.
+              // So we need to reset it here.
+              internalPattern: [],
 
-    '@typescript-eslint/brace-style': ['error', '1tbs', {
-      allowSingleLine: true
-    }],
-    '@typescript-eslint/comma-dangle': ['error', 'never'],
-    '@typescript-eslint/consistent-type-definitions': 'off',
-    '@typescript-eslint/consistent-type-imports': ['error', {
-      prefer: 'type-imports',
-      disallowTypeAnnotations: false,
-      fixStyle: 'inline-type-imports'
-    }],
-    '@typescript-eslint/no-empty-interface': 'off',
-    '@typescript-eslint/no-unused-vars': ['error', {
-      argsIgnorePattern: '^_',
-      destructuredArrayIgnorePattern: '^_'
-    }],
-    '@typescript-eslint/no-use-before-define': 'off',
+              customGroups: [
+                {
+                  groupName: 'aliased-external',
+                  elementNamePattern: '~.*/.*'
+                },
+                {
+                  groupName: 'parent-components',
+                  elementNamePattern: '@/components/.*'
+                },
+                {
+                  groupName: 'parent-composables',
+                  elementNamePattern: '@/composables/.*'
+                },
+                {
+                  groupName: 'aliased-internal',
+                  elementNamePattern: '@/.*'
+                }
+              ]
+            }
```

Comment:
not fully sure but why are components and composables separate than just `'@/.*'`? I don't think I need them on top of the other internal imports. I usually want them sorted by name. Not sure why it was earlier like that. Also, we don't use ~ imports anywhere. I think I'm fine with the default internalPattern.